### ISSUE TITLE
Generate documentation

### ISSUE THREAD

**Issue by Razaloc** [collabora/bmap-rs#48]
The project needs documentation. 
There are recommendations and standards for Rust crates documentation:

- [The rustdoc book](https://doc.rust-lang.org/rustdoc/how-to-write-documentation.html)
- [The Rust Programming Language](https://doc.rust-lang.org/book/ch14-02-publishing-to-crates-io.html)

**Comment by Razaloc** [comment]
Found out there is a [man crate](https://github.com/rust-cli/man) to allow the CLI application to have a man page. Wonder if we want to have one.

**Comment by obbardc** [comment]
> Found out there is a [man crate](https://github.com/rust-cli/man) to allow the CLI application to have a man page. Wonder if we want to have one.

Yes, that'd be great to have when we distribute as a Debian package. Can you check what other binary Rust Debian packages do ?
